group.go

```diff
@@ -32,6 +34,43 @@ func (g *Group) Generator() *big.Int {
 	return g.g
 }
 
+func (g *Group) MarshalBinary() (_ []byte, err error) {
+	var buf bytes.Buffer
+	enc := gob.NewEncoder(&buf)
+	// This array must be in the exact same order as the array used for unmarshalling.
```

Comment:
Thank you for that comment! That will help prevent otherwise easy to make errors in the future.